language/vm/src/serializer.rs

```diff
@@ -1088,6 +1098,19 @@ impl ModuleSerializer {
         Ok(())
     }
 
+    fn serialize_miscellaneous_items(
+        &mut self,
+        binary: &mut BinaryData,
+        module: &CompiledModuleMut,
+    ) -> Result<()> {
+        self.common.table_count = self.common.table_count.wrapping_add(1); // the count will bound to a small number
+        self.misc.0 = check_index_in_binary(binary.len())?;
+        // self module handle index
+        write_u16_as_uleb128(binary, module.self_module_handle_idx.0)?;
+        self.misc.1 = checked_calculate_table_size(binary, self.misc.0)?;
+        Ok(())
+    }
+
```

Comment:
@dariorussi Probably the only interesting part of this PR: currently I store the self module handle index in a new table called "MISC", but I'm open to change:
- Do we need a separate table for this? Would it be messy if we just store the field in another table like struct definitions?
- Better name if we do want a new table?

Reply:
I think we talked about this and for now let's go in and let's review soon.
Essentially the table is strange for a single element entry. 
Looking into making it an explicit "field" is probably the right thing